### DISCUSSION TITLE
transformerDirectives and transformerVariantGroup in Nuxt 3 config

### DISCUSSION THREAD

**Discussion by sandros94** [unocss/unocss#2402] [Q&A] [answered]
I'm wondering if it's just me doing this wrong or it isn't supported to have `transformerDirectives` and `transformerVariantGroup` inside the `nuxt.config.ts`?

Something like:
```ts [nuxt.config.ts]
export default defineNuxtConfig({
    modules: [
        '@unocss/nuxt',
        '@nuxtjs/color-mode',
        '@nuxt/image-edge',
        '@nuxt/content',
        '@nuxtjs/i18n',
        '@nuxt/devtools',
        '@vueuse/nuxt',
    ],
    css: [
        '@unocss/reset/tailwind.css',
    ],
    unocss: {
        uno: true,
        attributify: true,
        typography: true,
        preflight: true,
        icons: {
            cdn: 'https://esm.sh/',
              scale: 1.2,
              extraProperties: {
                'display': 'inline-block',
                'vertical-align': 'middle',
                // ...
            },
        },
        WebFonts: {
            fonts: {
                sans: 'DM Sans',
                serif: 'DM Serif Display',
                mono: 'DM Mono',
            },
        },
        // transformers applied here instead of unocss.config.ts
        transformers: [
            transformerDirectives(),
            transformerVariantGroup(),
        ],
    },
});
```
My aim was to create a nuxt boilerplate that I could point to from another nuxt project, using the [nuxt extend feature](https://nuxt.com/docs/api/configuration/nuxt-config#extends), and since I cannot extend the `unocss.config.ts` I though I could just move everything into my `nuxt.config.ts`.

P.S.: Yes I'm aware that I could add the imports for those two inside the `nuxt.config.ts`, but this brakes the extend feature.

**Comment by sandros94** [comment]
And I've just discovered that I cannot setup most of the config inside my `nuxt.config.ts` and then add just the transformers inside the `unocss.config.ts`. I have to use one or the other.

**Reply by sandros94** [reply]
I mean, if I'm not mistaken using `export default defineConfig` I'm overwriting the config I've made inside the `nuxt.config.ts`?
```ts unocss.config.ts
import {
    defineConfig,
    transformerDirectives,
    transformerVariantGroup,
} from 'unocss'

export default defineConfig({
    transformers: [
        transformerDirectives(),
        transformerVariantGroup(),
    ]
})
```

**Comment by zyyv** [comment, marked as answer]
Yes, it's not support `transformers` config in nuxt config.
https://github.com/unocss/unocss/blob/339f2b2c9be41a5505e7f4509eea1cf00a87a8d1/packages/nuxt/src/types.ts#L10-L81

The most recommended way is to use a dedicated `uno.config.ts` and put all configurations in it for the best development experience.

**Reply by sandros94** [reply]
In the end I'm thinking that I would mostly customize each nuxt instance using `uno.config.ts`, not being able to extend is not actually a problem per se (or not at all).
